code/modules/mob/living/carbon/slime/slime.dm

```diff
@@ -204,6 +204,12 @@
 	updatehealth()
 	return
 
+/mob/living/carbon/slime/MouseDrop(var/atom/movable/A as mob|obj)
+	if(isliving(A) && A != usr)
+		var/mob/living/Food = A
+		if(Food.Adjacent(usr) && !stat && Food.stat != DEAD) //messy
```

Comment:
you can replace everything here except the food.stat thing with canusetopic. it's centralized so you dont have to replace every single stat check everywhere when you add something new.